src/main/java/com/revlearn/team1/model/Course.java

```diff
@@ -8,57 +12,72 @@
 
 import java.time.LocalDate;
 import java.time.LocalDateTime;
+import java.util.ArrayList;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Set;
 
 @Entity
 @Data
 public class Course {
 
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
     @ManyToMany
     @JoinTable(
             name = "course_educators",
             joinColumns = @JoinColumn(name = "course_id"),
-            inverseJoinColumns = @JoinColumn(name = "educator_id")
+            inverseJoinColumns = @JoinColumn(name = "educator_id"),
+            uniqueConstraints = @UniqueConstraint(columnNames = {"course_id", "educator_id"})  // Ensure uniqueness
```

Comment:
you're enforcing unique constraints but use a list over a set. Might be a good idea to change that